src/carts/MagentoAddressHelper.js

```diff
@@ -79,29 +79,41 @@ class MagentoAddressHelper {
     * @return {Address}        CIF mapped address object
     */
     static mapToCIFAddress(magentoAddress) {
-        const address = new Address();
-        address.id = magentoAddress.id;
-        address.firstName = magentoAddress.firstname;
-        address.lastName = magentoAddress.lastname;
-        address.title = magentoAddress.prefix;
+        let streetName = null;
+        let streetNumber = null;
+        let additionalStreetInfo = null;
+        let additionalAddressInfo = null;
+
         if (Array.isArray(magentoAddress.street)) {
             if (magentoAddress.street[0]) {
-                address.streetName = magentoAddress.street[0];
+                streetName = magentoAddress.street[0];
             }
             if (magentoAddress.street[1]) {
-                address.streetNumber = magentoAddress.street[1];
+                streetNumber = magentoAddress.street[1];
             }
             if (magentoAddress.street[2]) {
-                address.additionalStreetInfo = magentoAddress.street[2];
+                additionalStreetInfo = magentoAddress.street[2];
             }
             if (magentoAddress.street[3]) {
-                address.additionalAddressInfo = magentoAddress.street[3];
+                additionalAddressInfo = magentoAddress.street[3];
             }
         }
-        address.postalCode = magentoAddress.postcode;
-        address.city = magentoAddress.city;
+
+        let address = new Address.Builder()
+            .withCity(magentoAddress.city)
+            .withCountry(magentoAddress.country_id)
+            .withFirstName(magentoAddress.firstname)
+            .withId(magentoAddress.id)
+            .withLastName(magentoAddress.lastname)
+            .withPostalCode(magentoAddress.postcode)
+            .withStreetName(streetName)
+            .build();
+
+        address.streetNumber = streetNumber;
```

Comment:
Why are these not set with the builder?

Reply:
I was wondering the same. IMO the builder should provide methods for all fields and the `build()` method is supposed to verify that all required fields are set.

Reply:
Not really, except that the important part was the required fields to avoid difficult bugs introduced by unwanted construction order through constructor. We can add that and validation in the build method but I would also leave it as a different topic as I see more impact on the code.

Reply:
How about leaving the validation for a different issue, but putting all properties in the builder now?

Reply:
I don't really see any added value for doing only this. I think it is better to have only required fields in the builder as long as it also serves as a way of identifying mandatory fields. In following releases, we can then add all fields to builders and validation warnings